### PR TITLE
Few merged changes and a temp commit of new order API

### PR DESCRIPTION
**Pull request by Firoz-PS**

### CODE REVIEW

File: resources/acceleron-server/routes/OrderRoute.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+let router = new ACCELERONCORE._routes.BaseRouter();
+let OrderController = require("../controllers/OrderController");
+
+router.post("/", async function (req, res, next) {
+  try {
+    const data = await new OrderController(req).newOrder();
```
Comment (abhijithcs) [inline resources/acceleron-server/routes/OrderRoute.js:6]: How different from:
KOTController(req).createNewKOT();

File: resources/acceleron-server/routes/KOTRoute.js [hunk 2]
```diff
@@ -19,6 +19,15 @@ router.get('/:id', async function (req, res, next) {
       }
 })
 
+router.post("/", async function (req, res, next) {
+  try {
+    const data = await new KOTController(req).createNewKOT();
```
Comment (abhijithcs) [inline resources/acceleron-server/routes/KOTRoute.js:24]: How different from:
OrderController(req).newOrder();

File: resources/acceleron-server/models/OrderModel.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+class OrderModel extends BaseModel {
+  constructor(request) {
+    super(request);
+    this.couch = ACCELERONCORE._connectors.CouchDBAsync;
+  }
+
+  async getTableStatus(table_req) {
+    const data = await this.couch
+      .get(
+        '/accelerate_kot/_design/kot-fetch/_view/fetchbytable?startkey=["' +
```
Comment (abhijithcs) [inline resources/acceleron-server/models/OrderModel.js:15]: Keep it in single line

**Review by abhijithcs — CHANGES_REQUESTED** [review]

File: resources/acceleron-server/services/OrderService.js [hunk 4]
```diff
@@ -0,0 +1,630 @@
...
+  async generateNewKOT(newOrderData) {
+    // hardcoded for now. to be removed once auth is implemented
+    var loggedInStaffInfoName = "Default";
+    var loggedInStaffInfoCode = "0000000000";
+    var accelerate_licencee_branch = "ADYAR";
+    var accelerate_licence_machineUID = "Z1500";
+
+    let data = {};
+
+    data = await this.SummaryService.getAllBillingParameters().catch((error) => {
```
Comment (abhijithcs) [inline resources/acceleron-server/services/OrderService.js:37]: Don't keep it data - may be more meaningful name say, bilingParamsData..

**Review by abhijithcs — CHANGES_REQUESTED** [review]

File: resources/acceleron-server/routes/OrderRoute.js [hunk 5]
```diff
@@ -19,6 +19,34 @@ router.put("/:id", async function (req, res, next) {
...
+router.post("/saved/new", async function (req, res, next) {
+  try {
+    const data = await new OrderController(req).saveOrder();
+    return await new BaseResponse(ResponseType.SUCCESS).send(res, data);
+  } catch (error) {
+    return next(error);
+  }
+});
+
+router.put("/saved/clear", async function (req, res, next) {
```
Comment (abhijithcs) [inline resources/acceleron-server/routes/OrderRoute.js:31]: /clearall

File: resources/acceleron-server/routes/OrderRoute.js [hunk 6]
```diff
@@ -19,6 +19,34 @@ router.put("/:id", async function (req, res, next) {
...
+router.put("/saved/:id/remove", async function (req, res, next) {
+  try {
+    const data = await new OrderController(req).removeSavedOrder();
+    return await new BaseResponse(ResponseType.SUCCESS).send(res, data);
+  } catch (error) {
+    return next(error);
+  }
+});
+
+// TO BE IMPLEMENTED
```
Comment (abhijithcs) [inline resources/acceleron-server/routes/OrderRoute.js:49]: Remove

File: resources/acceleron-server/routes/OrderRoute.js [hunk 7]
```diff
@@ -28,6 +56,7 @@ router.put("/:id/apply-discount", async function (req, res, next) {
   }
 });
 
+// TO BE IMPLEMENTED
```
Comment (abhijithcs) [inline resources/acceleron-server/routes/OrderRoute.js:59]: Remove

File: resources/acceleron-server/services/OrderService.js [hunk 8]
```diff
@@ -625,6 +627,71 @@ class OrderService extends BaseService {
       }
     }
   }
+
+  async saveOrder(newHoldingOrder) {
+    newHoldingOrder.timestamp = moment().format("HHmm");
```
Comment (abhijithcs) [inline resources/acceleron-server/services/OrderService.js:632]: Move moment().format("HHmm"); to TimeUtils.js

File: resources/acceleron-server/services/OrderService.js [hunk 9]
```diff
@@ -625,6 +627,71 @@ class OrderService extends BaseService {
...
+    let data = {}
+
+    try {
+      data = await this.SettingsService.addNewEntryToSettings("ACCELERATE_SAVED_ORDERS", newHoldingOrder);
+      //renderCustomerInfo()
```
Comment (abhijithcs) [inline resources/acceleron-server/services/OrderService.js:637]: Add //TODO

**Review by abhijithcs — CHANGES_REQUESTED** [review]

File: resources/acceleron-server/routes/SettingsRoute.js [hunk 10]
```diff
@@ -28,6 +28,15 @@ router.post("/:id/newentry", async function (req, res, next) {
   }
 });
 
+router.put("/:id", async function (req, res, next) {
```
[the remark this answers is not in the document]
Comment (Firoz-PS) [reply resources/acceleron-server/routes/SettingsRoute.js:31]: In OrderService for updating kot index and clearing all saved orders data

File: resources/acceleron-server/services/SettingsService.js [hunk 11]
```diff
@@ -37,6 +37,15 @@ class SettingsService extends BaseService {
     }
   }
 
+    async updateSettingsById(settings_id, new_entry) {
```
[the remark this answers is not in the document]
Comment (Firoz-PS) [reply resources/acceleron-server/services/SettingsService.js]: In OrderService for updating kot index and clearing all saved orders data

File: resources/acceleron-server/routes/OrderRoute.js (hunk 1 above)
Comment (Firoz-PS) [reply resources/acceleron-server/routes/OrderRoute.js:6]: This involves other preprocesses as well

File: resources/acceleron-server/routes/KOTRoute.js (hunk 2 above)
Comment (Firoz-PS) [reply resources/acceleron-server/routes/KOTRoute.js:24]: This doesn't involve the preprocesses for creating the kot object

**Review by abhijithcs — APPROVED** [review]

**Pull request merged**